src/OEmbed.php

```diff
@@ -53,11 +53,22 @@ protected function detectEndpoint(): ?UriInterface
     {
         $document = $this->extractor->getDocument();
 
-        return $document->link('alternate', ['type' => 'application/json+oembed'])
+        $endpoint = $document->link('alternate', ['type' => 'application/json+oembed'])
             ?: $document->link('alternate', ['type' => 'text/json+oembed'])
             ?: $document->link('alternate', ['type' => 'application/xml+oembed'])
             ?: $document->link('alternate', ['type' => 'text/xml+oembed'])
-            ?: $this->detectEndpointFromProviders();
+            ?: null;
+
+        if ($endpoint === null) {
+            return $this->detectEndpointFromProviders();
+        }
+
+        // Add configured OEmbed query parameters
+        parse_str($endpoint->getQuery(), $query);
+        $query = http_build_query(array_merge($query, $this->getOembedQueryParameters($query['url'])));
```

Comment:
I used the `array_merge` here to keep the original order of the parameters.

I don't think it's really necessary, but changing the parameters order invalidates some cached files and breaks some tests. If this solution is okay, I can also rename old cache files (I have saved myself that work for now).

Reply:
This overrides the query parameters of `$query` with the values returned by `getOembedQueryParameters` (including the values `url` and `format`). In think it's not needed to use this function, we can merge it directly with the settings:

```php
$query = array_merge($query, $this->extractor->getSetting('oembed:query_parameters') ?? []);
```

Reply:
Makes sense, I adjusted it. Thanks for the quick review!